### PR TITLE
Fixed get RC Applications

### PR DESCRIPTION
**Pull request by BSolonenko**
This PR is **[ready]** for review.

### Risk
This PR makes **[no]** API changes.

### Summary
- added is_rc_ field, getter and setter in ApplicationImpl;
- fixed RegisterAppInterfaceRequest::Run();
- fixed RCRPCPlugin::GetRCApplications().

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc [hunk 1]
```diff
@@ -340,21 +340,22 @@ void RegisterAppInterfaceRequest::Run() {
...
+      if (mobile_apis::AppHMIType::COMMUNICATION == current_app_type) {
         application->set_voice_communication_supported(true);
       }
-      if (mobile_apis::AppHMIType::PROJECTION ==
-          static_cast<mobile_apis::AppHMIType::eType>(
-              app_type.getElement(i).asUInt())) {
+      if (mobile_apis::AppHMIType::PROJECTION == current_app_type) {
         application->set_mobile_projection_enabled(true);
       }
+      if (mobile_apis::AppHMIType::REMOTE_CONTROL == current_app_type) {
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc]: @BSolonenko propose to replace all these if's with `switch (current_app_type)`

Comment (BSolonenko) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc]: @AKalinich-Luxoft  this enum contain value not handled in switch.

Comment (AKalinich-Luxoft) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc]: @BSolonenko so what is the problem? You can handle all needed HMI type values and add `default` case for all other values with an empty implementation

File: src/components/application_manager/include/application_manager/application.h [hunk 2]
```diff
@@ -443,6 +443,9 @@ class Application : public virtual InitialApplicationData,
   virtual bool is_navi() const = 0;
   virtual void set_is_navi(bool allow) = 0;
 
+  virtual bool is_rc() const = 0;
+  virtual void set_is_rc(bool allow) = 0;
```
Comment (AByzhynar) [inline src/components/application_manager/include/application_manager/application.h]: @BSolonenko parameter must be const

File: src/components/application_manager/include/application_manager/application_impl.h [hunk 3]
```diff
@@ -121,6 +121,11 @@ class ApplicationImpl : public virtual Application,
   }
   void set_is_navi(bool allow);
 
+  inline bool is_rc() const {
+    return is_rc_;
+  }
+  void set_is_rc(bool allow);
```
Comment (AByzhynar) [inline src/components/application_manager/include/application_manager/application_impl.h]: @BSolonenko 
Please add an empty line between functions

File: src/components/application_manager/include/application_manager/application_impl.h [hunk 4]
```diff
@@ -121,6 +121,11 @@ class ApplicationImpl : public virtual Application,
   }
   void set_is_navi(bool allow);
 
+  inline bool is_rc() const {
```
Comment (AByzhynar) [inline src/components/application_manager/include/application_manager/application_impl.h]: @BSolonenko This function breaks SDL coding style

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_rpc_plugin.cc [hunk 5]
```diff
@@ -61,17 +64,11 @@ RCRPCPlugin::Apps RCRPCPlugin::GetRCApplications(
...
-    return app ? app->QueryInterface(uid).valid() : false;
-  };
-
-  auto it = std::find_if(accessor.begin(), accessor.end(), predicate);
-
   std::vector<ApplicationSharedPtr> result;
-  while (it != accessor.end()) {
-    result.push_back(*it);
-    it = std::find_if(++it, accessor.end(), predicate);
+  for (auto& it : accessor) {
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_rpc_plugin.cc]: @BSolonenko Why not `const`?

File: src/components/application_manager/src/application_manager_impl.cc [hunk 6]
```diff
@@ -2458,7 +2458,7 @@ mobile_apis::Result::eType ApplicationManagerImpl::CheckPolicyPermissions(
     const std::string& function_id,
     const RPCParams& rpc_params,
     CommandParametersPermissions* params_permissions) {
-  LOG4CXX_INFO(logger_, "CheckPolicyPermissions");
+  LOG4CXX_AUTO_TRACE(logger_);
   // TODO(AOleynik): Remove check of policy_enable, when this flag will be
   // unused in config file
```
Comment (AByzhynar) [inline src/components/application_manager/src/application_manager_impl.cc]: @BSolonenko Check if we still need this comment

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc (hunk 1 above)
Comment (BSolonenko) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc]: @AKalinich-Luxoft Fixed in https://github.com/smartdevicelink/sdl_core/pull/2048/commits/3ffd281451b051e9ead29c3a877a6678109321ac

File: src/components/application_manager/include/application_manager/application.h (hunk 2 above)
Comment (BSolonenko) [reply src/components/application_manager/include/application_manager/application.h]: @AByzhynar  Fixed in https://github.com/smartdevicelink/sdl_core/pull/2048/commits/3ffd281451b051e9ead29c3a877a6678109321ac

File: src/components/application_manager/include/application_manager/application_impl.h (hunk 3 above)
Comment (BSolonenko) [reply src/components/application_manager/include/application_manager/application_impl.h]: @AByzhynar Fixed in https://github.com/smartdevicelink/sdl_core/pull/2048/commits/3ffd281451b051e9ead29c3a877a6678109321ac

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_rpc_plugin.cc (hunk 5 above)
Comment (BSolonenko) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_rpc_plugin.cc]: @AByzhynar  fixed in https://github.com/smartdevicelink/sdl_core/pull/2048/commits/3ffd281451b051e9ead29c3a877a6678109321ac

File: src/components/application_manager/include/application_manager/application.h [hunk 7]
```diff
@@ -443,6 +443,17 @@ class Application : public virtual InitialApplicationData,
   virtual bool is_navi() const = 0;
   virtual void set_is_navi(bool allow) = 0;
 
+  /**
+   * @brief Returns is_rc
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/include/application_manager/application.h]: @BSolonenko unclear description. Please update it

Comment (AKalinich-Luxoft) [reply src/components/application_manager/include/application_manager/application.h]: @BSolonenko also please add `@return` line with description

File: src/components/application_manager/include/application_manager/application.h [hunk 8]
```diff
@@ -443,6 +443,17 @@ class Application : public virtual InitialApplicationData,
...
+   */
+  virtual bool is_remote_control_supported() const = 0;
+
+  /**
+   * @brief Sets is_rc, which is used to determine rc app
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/include/application_manager/application.h]: @BSolonenko please update description

File: src/components/application_manager/include/application_manager/application.h (hunk 7 above)
Comment (BSolonenko) [reply src/components/application_manager/include/application_manager/application.h]: @AKalinich-Luxoft  Fixed in https://github.com/smartdevicelink/sdl_core/pull/2048/commits/89790bbec52b1c03e17e863389ee538708ebf277

File: src/components/application_manager/include/application_manager/application.h (hunk 8 above)
Comment (BSolonenko) [reply src/components/application_manager/include/application_manager/application.h]: @AKalinich-Luxoft  Fixed in https://github.com/smartdevicelink/sdl_core/pull/2048/commits/89790bbec52b1c03e17e863389ee538708ebf277

File: src/components/application_manager/src/application_manager_impl.cc (hunk 6 above)
Comment (BSolonenko) [reply src/components/application_manager/src/application_manager_impl.cc]: @AByzhynar  policy_enable used in 'smartDeviceLink.ini'

File: src/components/application_manager/include/application_manager/application.h [hunk 9]
```diff
@@ -443,6 +443,19 @@ class Application : public virtual InitialApplicationData,
...
+  /**
+   * @brief Returns is_remote_control_supported_
+   * @return true if app supports remote control, else false
+   */
+  virtual bool is_remote_control_supported() const = 0;
+
+  /**
+   * @brief Sets remote control supported,
+   * which is used to determine app with remote control
+   * @param allow
```
Comment (AByzhynar) [inline src/components/application_manager/include/application_manager/application.h]: @BSolonenko parameter description missed

**Pull request merged**